msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AuthScheme.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.aad.msal4j;
+
+import java.net.URI;
+
+//TODO: javadocs explaining this class, link to documentation page
+public class AuthScheme {
```

Comment:
Is it necessary to expose a the "Auth Scheme" concept? The only API we need is 

`WithProofOfPosession(Uri uri, HttpMethod m, string nonce)`

Reply:
Internally you can use this concept, but MSAL APIs do not expose it. 

The spec is here - https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=/PoPTokensProtocol/PoP_API_In_MSAL.md&version=GBdev&anchor=key-case-sensitivity

You should also implement a helper named `IsProofOfPosessionSupportedByClient`

Reply:
Originally I just followed the design in [Python's PR](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/511), since it also introduced the concepts of PoP tokens, auth schemes, etc., and that seemed to just have `AuthScheme` as the parameter instead of `WithProofOfPosession` so I figured that was just the minimal design we should aim for too.

I agree it should be more consistent with the design doc though, so in the latest commit the relevant flows have a parameter named `proofOfPossession` (our parameters don't start with 'with'), this `AuthScheme` class was renamed to PopParameters (since we'd only have the one scheme anyway) and it's package-private.